app/models/user.rb

```diff
@@ -1,2 +1,6 @@
 class User < ApplicationRecord
+  def by_level(level)
+    Test.joins('join test_passages on test_passages.test_id = tests.id')
+        .where('test_passages.user_id = ? and tests.level = ?', self.id, level)
```

Comment:
```suggestion
        .where(test_passages: { user_id: id },  tests: { level: level})
```
рекомендуется по возможности писать через хеши условия
проще читать, проще изменять